### ISSUE TITLE
BUG: pdist docs are unclear

### ISSUE THREAD

**Issue by Tim-Kirkwood** [scipy/scipy#15337]
### Describe your issue.

The pdist input array is described as `An m by n array of m original observations in an n-dimensional space` but pdist [appears to expect a boolean array](https://stackoverflow.com/a/35530334/11357695) for at least one metric (jaccard) based on pdist activity.  If that's correct, the docs should clarify that different metrics require different input matricies (and/or a processed matrix) rather than universally expecting a matrix of original observations.

### Reproducing Code Example

```python
NA
```

### Error message

```shell
NA
```

### SciPy/NumPy/Python version information

NA

**Comment by tylerjereddy** [comment]
> rather than universally expecting a matrix of original observations.

I agree that a doc improvement might be possible. Can't bools be original observations as well though? There probably are type limitations for other metrics as well if you try to feed in "complex" numbers or stuff like that for some measures. Perhaps suggesting that users check the docstrings of the individual metric they are using for more details or something like that.

**Comment by Tim-Kirkwood** [comment]
Hello, apologies I was actually incorrect.  [pdist(metric = 'jaccard')](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html) doesn't need to take an array of bools, unlike the main scipy jaccard function, but it does compare arrays [index by index](https://stackoverflow.com/a/70553656/11357695) rather than by comparing the list as a whole, which is what caused my confusion.  I think this is covered by "Given two vectors, u and v, the Jaccard distance is the proportion of those elements u[i] and v[i] that disagree", but I think this could still be clearer - perhaps simply adding an example for each measure (as is done for the individual distance measures like [Jaccard](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.jaccard.html#scipy.spatial.distance.jaccard) would be a good idea?  

Simply referring to a docstring wouldn't be ideal as the behaviour of pdist(metric='jaccard'), which in fact does not need booleans, is different to that of jaccard, which does.
